mmv1/third_party/terraform/services/cloudidentity/resource_cloud_identity_group_test.go.erb

```diff
@@ -177,3 +185,87 @@ func testAccCheckCloudIdentityGroupDestroyProducer(t *testing.T) func(s *terrafo
 		return nil
 	}
 }
+
+func TestAccCloudIdentityGroup_cloudIdentityGroupsDynamicExample(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"org_domain":    envvar.GetTestOrgDomainFromEnv(t),
+		"cust_id":       envvar.GetTestCustIdFromEnv(t),
+		"random_suffix": acctest.RandString(t, 10),
+	}
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck:     func() { acctest.AccTestPreCheck(t) },
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy: testAccCheckCloudIdentityGroupDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCloudIdentityGroup_cloudIdentityGroupsDynamicExample(context),
+			},
+			{
+				ResourceName:            "google_cloud_identity_group.cloud_identity_groups_dynamic",
+				ImportState:             true,
+				ImportStateVerify:       true,
+			},
+			{
+				Config: testAccCloudIdentityGroup_cloudIdentityGroupsDynamicUpdate(context),
+			},
+			{
+				ResourceName:            "google_cloud_identity_group.cloud_identity_groups_dynamic",
+				ImportState:             true,
+				ImportStateVerify:       true,
+			},
+		},
+	})
+}
+
+func testAccCloudIdentityGroup_cloudIdentityGroupsDynamicExample(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_cloud_identity_group" "cloud_identity_groups_dynamic" {
+  display_name         = "tf-test-my-identity-dynamic-group%{random_suffix}"
+
+  parent = "customers/%{cust_id}"
+
+  group_key {
+    id = "tf-test-my-identity-dynamic-group%{random_suffix}@%{org_domain}"
+  }
+
+  labels = {
+    "cloudidentity.googleapis.com/groups.discussion_forum" = ""
+  }
+
+  dynamic_group_metadata {
+    queries {
+      resource_type = "USER"
+      query         = "user.addresses.exists(ad, ad.locality=='Sunnyvale')"
+    }
+  }
+}
+`, context)
+}
+
+func testAccCloudIdentityGroup_cloudIdentityGroupsDynamicUpdate(context map[string]interface{}) string {
+	return acctest.Nprintf(`
+resource "google_cloud_identity_group" "cloud_identity_groups_dynamic" {
+  display_name         = "tf-test-my-identity-dynamic-group%{random_suffix}"
+
+  parent = "customers/%{cust_id}"
+
+  group_key {
+    id = "tf-test-my-identity-dynamic-group%{random_suffix}@%{org_domain}"
+  }
+
+  labels = {
```

Comment:
Looks like the automatically added dynamic group label is causing problems here. We may need to either add a DiffSuppressFunc for that label, or add it explicitly to the tests

Reply:
Thanks, added it to the tests.